### DISCUSSION TITLE
[Help]: How do I get the next scroll snap?

### DISCUSSION THREAD

**Discussion by erkanaltinors** [davidjerleke/embla-carousel#1089] [Help] [answered]
### Which variants of Embla Carousel are relevant to this feature request?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [X] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Feature description

previousScrollSnap method returns the previous item index but there is missing method to get next item's index.

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for feature requests

**Comment by davidjerleke** [comment, marked as answer]
@erkanaltinors you can get the next scroll snap index like so:

```tsx
const nextSnapIndex = emblaApi.internalEngine().index.add(1).get();
```

**Reply by erkanaltinors** [reply]
thank you @davidjerleke !

works like a charm.

I used this method to make it compatible with visible carousel items length as below:

`api.internalEngine().index.add(api.slidesInView().length - 1).get()`
